actionpack/lib/action_dispatch/routing/url_for.rb

```diff
@@ -147,6 +149,24 @@ def url_options
       #    # => 'http://somehost.org/myapp/tasks/testing'
       #    url_for controller: 'tasks', action: 'testing', host: 'somehost.org', script_name: "/myapp", only_path: true
       #    # => '/myapp/tasks/testing'
+      #
+      # Missing Routes keys may be filled in from the current request's parameters
+      # (e.g. +:controller+, +:action+, +:id+ and any other parameters that are
+      # placed in the path). This allows one to have the following code:
+      #
+      #   class UsersController < Rails::Application
```

Comment:
I guess you meant `ApplicationController` instead of `Rails::Application`. Anyway, I'm not sure whether creating a controller is really useful here ; what do you think about putting a sentence before the code examples like:

> Given that the current action has been reached through `GET /users/1`: